sources/packages/web/src/components/common/OverawardDetails.vue

```diff
@@ -151,45 +118,51 @@ export default defineComponent({
       useFormatters();
     const snackBar = useSnackBar();
     const { mobile: isMobile } = useDisplay();
+    const isAddition = ref(true);
 
     const overawardDetails = ref([] as OverawardAPIOutDTO[]);
-    const addManualOverawardDeduction = ref(
+    const addManualOveraward = ref(
       {} as ModalDialog<OverawardManualRecordAPIInDTO | boolean>,
     );
-    const overawardDeductionOriginTypes = new Set([
-      DisbursementOverawardOriginType.AwardDeducted,
-      DisbursementOverawardOriginType.ManualRecord,
-    ]);
-    const overawards = computed(() =>
-      overawardDetails.value.filter(
-        (overaward) =>
-          !overawardDeductionOriginTypes.has(overaward.overawardOrigin),
-      ),
-    );
-
-    const overawardDeductions = computed(() =>
-      overawardDetails.value.filter((overaward) =>
-        overawardDeductionOriginTypes.has(overaward.overawardOrigin),
-      ),
-    );
+    const addOverawards = async () => {
+      isAddition.value = true;
+      const manualOveraward = await addManualOveraward.value.showModal();
+      if (!manualOveraward || typeof manualOveraward === "boolean") {
+        return;
+      }
+      try {
+        await OverawardService.shared.addManualOveraward(
+          props.studentId as number,
+          manualOveraward,
+        );
+        snackBar.success("Overaward added successfully.");
+        context.emit("manualOverawardAdded");
+        await loadOverawardDetails();
+      } catch {
+        snackBar.error("An error happened while adding overaward.");
+      }
+    };
 
-    const addManualOveraward = async () => {
-      const manualOveraward =
-        await addManualOverawardDeduction.value.showModal();
-      if (manualOveraward) {
-        try {
-          await OverawardService.shared.addManualOverawardDeduction(
-            props.studentId as number,
-            manualOveraward as OverawardManualRecordAPIInDTO,
-          );
-          snackBar.success("Overaward deduction added successfully.");
-          context.emit("manualOverawardAdded");
-          await loadOverawardDetails();
-        } catch {
-          snackBar.error(
-            "An error happened while adding manual overaward deduction.",
-          );
-        }
+    const subtractOverawards = async () => {
+      isAddition.value = false;
+      const manualOveraward = await addManualOveraward.value.showModal();
+      if (!manualOveraward || typeof manualOveraward === "boolean") {
+        return;
+      }
+      try {
+        const subtractOveraward = {
+          ...manualOveraward,
+          overawardValue: -manualOveraward.overawardValue,
+        };
+        await OverawardService.shared.addManualOveraward(
+          props.studentId as number,
+          subtractOveraward,
+        );
+        snackBar.success("Overaward subtracted successfully.");
+        context.emit("manualOverawardAdded");
+        await loadOverawardDetails();
```

Comment:
I'd also move the "loadOverawardDetails" outside the try/catch block, or you can use "finally". loadOverawardDetails should have its own error handling

Reply:
I'll leave it in the same location but add error handling to loadOverawardDetails. This way the source of the error will be more specific without multiple errors being displayed.